tests/components/lg_soundbar/test_config_flow.py

```diff
@@ -209,6 +208,208 @@ def temescal_side_effect(addr, port, callback):
+    mock_uuid_q.empty.assert_called_once()
+    mock_uuid_q.put_nowait.assert_not_called()
+    mock_uuid_q.get.assert_called_once()
+
+
+async def test_form_both_queues_empty(hass):
+    """Test we get the form, but none of the data we want is provided by the API."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    mock_uuid_q = MagicMock()
+    mock_name_q = MagicMock()
+
+    with patch(
+        "homeassistant.components.lg_soundbar.config_flow.temescal"
+    ) as mock_temescal, patch(
+        "homeassistant.components.lg_soundbar.config_flow.Queue"
+    ) as mock_q, patch(
+        "homeassistant.components.lg_soundbar.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        mock_q.side_effect = [mock_uuid_q, mock_name_q]
+        mock_uuid_q.empty.return_value = True
+        mock_name_q.get.side_effect = Empty()
+
+        tmock = mock_temescal.temescal
+        tmock.return_value = Mock()
+        instance = tmock.return_value
+
+        def temescal_side_effect(addr, port, callback):
+            mac_info = {"msg": "MAC_INFO_DEV"}
+            instance.get_mac_info.side_effect = lambda: callback(mac_info)
+            product_info = {"msg": "PRODUCT_INFO"}
+            instance.get_product_info.side_effect = lambda: callback(product_info)
+            info = {"msg": "SPK_LIST_VIEW_INFO"}
+            instance.get_info.side_effect = lambda: callback(info)
+            return DEFAULT
+
+        tmock.side_effect = temescal_side_effect
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: "1.1.1.1",
+            },
+        )
+        await hass.async_block_till_done()
+
+    print("result2 " + str(result2))
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "no_data"}
+    assert len(mock_setup_entry.mock_calls) == 0
+    mock_uuid_q.empty.assert_called_once()
+    mock_uuid_q.put_nowait.assert_not_called()
+    mock_name_q.put_nowait.assert_not_called()
+    mock_name_q.get.assert_called_once()
+
+
+async def test_no_uuid_host_already_configured(hass):
+    """Test we handle if the device has no UUID and the host has already been configured."""
+
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_HOST: "1.1.1.1",
+            CONF_PORT: DEFAULT_PORT,
+        },
+    )
+    mock_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch("homeassistant.components.lg_soundbar.config_flow.temescal"), patch(
+        "homeassistant.components.lg_soundbar.config_flow.test_connect",
```

Comment:
Don't patch the integration code in the config flow. We want to aim at patching the library and as few parts of our code in the config flow as possible.